pkg/cmd/provisioning/ibmcloud/create_test.go

```diff
@@ -0,0 +1,103 @@
+package ibmcloud
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/openshift/cloud-credential-operator/pkg/cmd/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	apiKey        = "testapiKey"
+	testDirPrefix = "createtestdir"
+)
+
+func TestIAMRoles(t *testing.T) {
+	tests := []struct {
+		name        string
+		setup       func(*testing.T) string
+		verify      func(t *testing.T, tempDirName string)
+		cleanup     func(*testing.T)
+		expectError bool
+	}{
+		{
+			name: "Generate Secret for one CredentialsRequest",
+			setup: func(t *testing.T) string {
+				tempDirName, err := ioutil.TempDir(os.TempDir(), testDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory")
+
+				err = testCredentialsRequest(t, "firstcredreq", "namespace1", "secretName1", tempDirName)
+				require.NoError(t, err, "Errored while setting up test CredReq files")
+
+				return tempDirName
+			},
+			verify: func(t *testing.T, targetDir string) {
+				files, err := ioutil.ReadDir(targetDir)
+				require.NoError(t, err, "Unexpected error listing files in targetDir")
+
+				assert.Equal(t, 1, len(files), "Should be exactly 1 Secret generated for 1 CredentialsRequest")
+			},
+			cleanup: func(t *testing.T) {
+				return
+			},
+			expectError: false,
```

Comment:
Can we have a test that does setting IC_API_KEY in setup and verify if generated secret actually has that key? Also, one negative test where env var is not set and error is thrown.